### DISCUSSION TITLE
Problems scaling when using different start.degree & gap.after for tracks.

### DISCUSSION THREAD

**Discussion by stephenwhitmarsh** [jokergoo/circlize#285] [Q&A]
Dear Zuguang Gu (or users)

First of all my compliments! I love the Circlize package and am planning to use it for my publications.

I am afraid, however, that I might be using it a bit beyond as intended, or in any case I've been struggling a lot with plotting segments that have different offsets (`start.degree` & `gap.after`) for individual tracks. Following the example in the book, I've been trying to use the workaround with `par(new = TRUE)`. However, since this necessitates the use of `canvas.xlim` and `canvas.ylim`, I am not having the benefit of automatically scaled tracks (distance between them, while maintaining the same height), as you can see in the attached picture. I've come close, but not close enough with the following code (see first two line):

```
track_height = 0.1 * 1.2 ^ c(1,2,3) 
plot_size    = 1 * 1.2 ^ c(1,2,3)
dev.off()

for(ipart in 1:3) {
  
  circos.par("track.height" = track_height[ipart], 
             "start.degree" = 90+360-position_sel$offset[ipart]*360, 
             "gap.after" = 360-position_sel$duration[ipart]*360,
             "canvas.xlim" = c(-plot_size[ipart], plot_size[ipart]), 
             "canvas.ylim" = c(-plot_size[ipart], plot_size[ipart]))
  circos.initialize(hyp_sel$dummy[hyp_sel$patient == 1], x = hyp_sel$dt[hyp_sel$patient == 1])
  circos.track(hyp_sel$part[hyp_sel$patient == 1], ylim = c(1, 5))
  circos.lines(hyp_sel$dt[hyp_sel$patient == 1], as.numeric(hyp_sel$label[hyp_sel$patient == 1], type = 's' ))
  circos.clear()
  
  par(new = TRUE) # <- magic
}
```
I'd like to scale up the number of tracks, so not being able to tightly organize the tracks becomes increasingly problematic, especially when also trying out differnt heights of the tracks (all the same though). Would you know how to deal with this either algorithmically (such as my first two lines), or perhaps I could try another, more intended method. Ideally, I would just be able to add tracks as normal, and then shift them (`start.degree` & `gap.after`) when/after I add them.

Any help would be greatly appreciated!

![circlize](https://user-images.githubusercontent.com/12006406/138130403-8c3c1933-ff07-44e1-8d5c-e5b7b088db24.png)

**Comment by jokergoo** [comment]
I think you don't need to directly manipulate `canvas.xlim`/`canvas.ylim`. That is too confusing and makes things complicated. For your use case, I would suggest creating one single sector that covers the whole circle, and in each track, you draw rectangles with different lengths, to simulate tracks with different length:

```r
circos.initialize(sectors = "foo", xlim = c(0, 1))
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
	circos.rect(0, 0, 0.6, 1)  # pretend this is a track
	x = seq(0, 0.6, length = 100)
	y = runif(100)
	circos.lines(x, y, type = "s", area = TRUE)
}, bg.border = NA, track.height = 0.1)

circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
	circos.rect(0, 0, 0.8, 1)
}, bg.border = NA, track.height = 0.1)

circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
	circos.rect(0, 0, 0.4, 1)
}, bg.border = NA, track.height = 0.1)
```

![image](https://user-images.githubusercontent.com/449218/140891940-6275c4f2-bd55-49e4-9bad-25bea8c303ed.png)
